source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/BaseActionElementRenderer.java

```diff
@@ -266,47 +207,136 @@ private Activity getActivity(Context context)
             return null;
         }
 
-        @Override
-        public void onClick(View v) {
-            if (m_isInlineShowCardAction)
+        private void populateViewsDictionary()
+        {
+            m_viewDictionary = new HashMap<>();
+
+            ToggleVisibilityTargetVector toggleVisibilityTargetVector = m_toggleVisibilityAction.GetTargetElements();
+            View rootView = m_renderedAdaptiveCard.getView();
+
+            for (int i = 0; i < toggleVisibilityTargetVector.size(); ++i)
             {
-                Activity hostingActivity = getActivity(v.getContext());
-                if (hostingActivity != null) {
-                    View currentFocusedView = hostingActivity.getCurrentFocus();
-                    if (currentFocusedView != null) {
-                        currentFocusedView.clearFocus();
-                    }
+                ToggleVisibilityTarget target = toggleVisibilityTargetVector.get(i);
+                String elementId = target.GetElementId();
+
+                View foundView = rootView.findViewWithTag(new TagContent(elementId));
+                if (foundView != null)
```

Comment:
do you expect to not find the target element in the rootView object? What are the cases where this is not null?

Reply:
if the card creator were to toggle the visibility of an element that doesn't exist (giving an id that doesn't exist for any element in the card) then the found view will be null and as such we can't change the visibility for that view

---
In reply to: [277106481](https://github.com/Microsoft/AdaptiveCards/pull/2717#discussion_r277106481) [](ancestors = 277106481)